src/integration/transaction/application/services/transaction.service.ts

```diff
@@ -1,66 +1,86 @@
 import { Injectable, NotFoundException } from '@nestjs/common';
 import { CTransactionSegWit } from '@defichain/jellyfish-transaction';
 import { RawTxDto } from 'src/blockchain/ain/jellyfish/dto/raw-tx.dto';
-import { TransactionDto } from '../dto/transaction.dto';
+import { TransactionOutputDto } from '../dto/transaction.dto';
 import { Transaction } from '../types/transaction';
-import { TxCheck } from '../util/tx-check';
+import { Transaction as TransactionEntity } from '../../domain/entities/transaction.entity';
 import { SmartBuffer } from 'smart-buffer';
+import { TransactionRepository } from '../repositories/transaction.repository';
+import { Config } from 'src/config/config';
 
 @Injectable()
 export class TransactionService {
   private readonly transactions: Map<string, Transaction>;
 
-  constructor() {
+  constructor(private readonly repository: TransactionRepository) {
     this.transactions = new Map<string, Transaction>();
   }
 
-  getOpen(): TransactionDto[] {
-    return this.select((tx: TransactionDto) => TxCheck.isOpen(tx));
+  async getOpen(): Promise<TransactionOutputDto[]> {
+    return this.repository.getOpen().then(this.toDto);
   }
 
-  getVerified(): TransactionDto[] {
-    return this.select((tx: TransactionDto) => TxCheck.isVerified(tx));
+  async getVerified(): Promise<TransactionOutputDto[]> {
+    return this.repository.getVerified().then(this.toDto);
   }
 
-  verified(id: string, signature: string) {
-    const tx = this.transactions.get(id);
+  async verified(id: string, signature: string) {
+    const tx = await this.repository.findOne(id);
     if (!tx) throw new NotFoundException('Transaction not found');
-    tx.verifierSignature = signature;
+    this.repository.save(tx.verified(signature));
   }
 
-  invalidated(id: string, reason?: string) {
-    const tx = this.transactions.get(id);
+  async invalidated(id: string, reason?: string) {
+    const tx = await this.repository.findOne(id);
+    const txPromise = this.transactions.get(tx.id);
     if (!tx) throw new NotFoundException('Transaction not found');
-    console.warn(`Invalidated ${id} with reason: ${reason}`);
     this.transactions.delete(tx.id);
-    tx.invalidated();
+    txPromise?.invalidated();
+    this.repository.save(tx.invalidated(reason));
   }
```

Comment:
One could split the logic into map and DB related things (for all update methods), I don't know if it makes it more readable ... up to you 😉 
```ts
  async invalidated(id: string, reason?: string) {
    const tx = await this.repository.findOne(id);
    if (!tx) throw new NotFoundException('Transaction not found');
    this.repository.save(tx.invalidated(reason));

    const txPromise = this.transactions.get(tx.id);
    this.transactions.delete(tx.id);
    txPromise?.invalidated();
  }
```

Reply:
yeah it does. I will never write code friday evening while talking to a lot of people about non related topics

Reply:
Haha 🤣